sources/api/netdog/src/net_config/devices/bonding.rs

```diff
@@ -0,0 +1,161 @@
+                }
+            )
+        }
+        // Validate monitoring configuration
+        match &self.monitoring_config {
+            BondMonitoringConfig::MiiMon(config) => config.validate()?,
+            BondMonitoringConfig::ArpMon(config) => config.validate()?,
+        }
+
+        Ok(())
+    }
+}
+
+// Currently only mode 1 (active-backup) is supported but eventually 0-6 could be added
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) enum BondMode {
+    ActiveBackup,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(untagged)]
+pub(crate) enum BondMonitoringConfig {
+    MiiMon(MiiMonitoringConfig),
+    ArpMon(ArpMonitoringConfig),
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct MiiMonitoringConfig {
+    #[serde(rename = "miimon-frequency-ms")]
+    pub(crate) frequency: u32,
+    #[serde(rename = "miimon-updelay-ms")]
+    pub(crate) updelay: u32,
+    #[serde(rename = "miimon-downdelay-ms")]
+    pub(crate) downdelay: u32,
+}
+
+impl Validate for MiiMonitoringConfig {
+    fn validate(&self) -> Result<()> {
+        ensure!(
+            self.frequency > 0,
+            error::InvalidNetConfigSnafu {
+                reason: "miimon-frequency-ms of 0 disables Mii Monitoring, either set a value or configure Arp Monitoring"
+            }
+        );
+        // updelay and downdelay should be a multiple of frequency, but will be rounded down
+        // by the kernel, this ensures they are at least the size of frequency (non-zero)
+        ensure!(
+            self.frequency <= self.updelay && self.frequency <= self.downdelay,
+            error::InvalidNetConfigSnafu {
+                reason: "miimon-updelay-ms and miimon-downdelay-ms must be equal to or larger than miimon-frequency-ms"
+            }
+        );
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct ArpMonitoringConfig {
+    #[serde(rename = "arpmon-interval-ms")]
+    pub(crate) interval: u32,
+    #[serde(rename = "arpmon-validate")]
+    pub(crate) validate: ArpValidate,
+    #[serde(rename = "arpmon-targets")]
+    pub(crate) targets: Vec<IpAddr>,
+}
+
+impl Validate for ArpMonitoringConfig {
+    fn validate(&self) -> Result<()> {
+        ensure!(
+            self.interval > 0,
+            error::InvalidNetConfigSnafu {
+                reason: "arpmon-interval-ms of 0 disables Arp Monitoring, either set a value or configure Mii Monitoring"
+            }
+        );
+        // If using Arp Monitoring, 1-16 targets must be specified
+        let targets_length: u32 = self.targets.len() as u32;
+        ensure!(
+            targets_length > 0 && targets_length <= 16,
+            error::InvalidNetConfigSnafu {
+                reason: "arpmon-targets must include between 1 and 16 targets"
+            }
+        );
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
```

Comment:
There are probably a few enums like this in the PR. It can be handy when writing code if "simple" enums implement `Copy` since they are nothing but an integer under the hood.

```suggestion
#[derive(Clone, Copy, Debug, Deserialize)]
```

Reply:
Ok, I'll take a look at adding them to the Enums that are "simple" like you call out.